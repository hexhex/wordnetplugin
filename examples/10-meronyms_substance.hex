%% this returns all (substance) meronyms of a word
%% POS: nouns

%% the search number for (substance) meronyms
%% is 10.
%% Note: In the WordNet documentation this is
%%       the search number for substance holonyms.
%%       In fact search 10 fetches words, for
%%       which *word* is a holonym (and these
%%       are its meronyms)

word("water").
meronyms(W,SW,M,SM) :- &wordnet[10,W](1,SW,M,SM), word(W).

meronyms("water",1,"oxygen",2), 
meronyms("water",1,"hydrogen",1), 
meronyms("water",2,"water",1)?
