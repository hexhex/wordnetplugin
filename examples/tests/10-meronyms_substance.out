{meronyms("water",1,"atomic_number_8",2), meronyms("water",1,"O",2), meronyms("water",1,"oxygen",2), meronyms("water",1,"atomic_number_1",1), meronyms("water",1,"H",1), meronyms("water",1,"hydrogen",1), meronyms("water",2,"H2O",1), meronyms("water",2,"water",1), word("water")}
